packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -268,6 +269,13 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// Defaults to being adapted to the current [TargetPlatform].
   final bool centerTitle;
 
+  /// The margin between title and leading content. This margin is applied
+  /// even if there is no leading content. If you want title to take all the
+  /// space available, set this value to 0.0.
+  ///
+  /// Defaults to the specification in [NavigationToolbar].
```

Comment:
I'm worried about overloading the semantics behind "margin" here. I've always thought about "margin" as something that never pushes you off the walls, but only off siblings, _if any_. To push off the wall you'd use "padding". If we go with this design, I'd use a different word, perhaps "spacing"?

That said, I think we can also fix this issue with `padding` as something that pushes all content off the wall irrespective of which children are present.

Reply:
I like spacing.

TBH, margin confused me initially as well which is why I wanted to remove it in the first place :-).

Reply:
titleSpacing seems fine to me.

Reply:
(but i agree that "margin" is weird)